src/Helper/Helper.php

```diff
@@ -68,18 +93,198 @@ public function getSubmissionIdFromJob(int $jobId): ?int {
   }
 
   /**
-   * Get all jobs from advanced queue table.
+   * Get all jobs that match a specific form.
+   *
+   * @param string $formId
+   *   The form to match.
    *
    * @return array
-   *   A list of jobs.
+   *   A list of view parameters.
    *
-   * @phpstan-return array<int, \stdClass>
+   * @phpstan-return array<string, mixed>
    */
-  public function getAllJobs():array {
-    $query = $this->connection->select('advancedqueue', 'a');
-    $query->fields('a', ['payload', 'job_id']);
+  public function getQueueJobIds(string $formId): array {
+    $query = $this->connection->select('os2forms_queue_submission_relation', 'o');
+    $query->fields('o', ['job_id']);
+    $query->condition('webform_id', $formId, '=');
+
+    return $query->execute()->fetchCol();
+  }
+
+  /**
+   * Handle a job from the AdvancedQueueProcessSubscriber.
+   *
+   * @param \Drupal\advancedqueue\Job $job
+   *   The job about to be processed.
+   */
+  public function handleJob(Job $job): void {
+    $data = $this->getDataFromPayload($job->getPayload());
+    if ($data) {
+      $data['job_id'] = (int) $job->getId();
+      $this->addRelation($data);
+    }
+  }
+
+  /**
+   * Handle importing advanced queue jobs through a command.
+   */
+  public function handleImport(): void {
+    $jobs = $this->getAllQueueJobs();
+    foreach ($jobs as $job) {
+      $this->handleJob($job);
+    }
+  }
+
+  /**
+   * Get webform id from job queue id.
+   *
+   * @param string $jobId
+   *   The id of a queue job.
+   *
+   * @return string
+   *   The webform id.
+   */
+  public function getWebformIdFromQueue(string $jobId): string {
+    $query = $this->connection->select('os2forms_queue_submission_relation', 'o');
+    $query->condition('job_id', $jobId, '=');
+    $query->fields('o', ['webform_id']);
+
+    return $query->execute()->fetchObject()->webform_id;
+  }
+
+  /**
+   * Retrieve data from advanced queue job.
+   *
+   * @param array $payload
+   *   The payload from an advanced queue job.
+   *
+   * @return array|null
+   *   An array containing submission id and webform id.
+   *
+   * @phpstan-param array<string, mixed> $payload
+   * @phpstan-return array<string, mixed>
+   */
+  private function getDataFromPayload(array $payload): ?array {
+    $submissionId = $this->getSubmissionId($payload);
+    if (empty($submissionId)) {
+      return NULL;
+    }
+
+    try {
+      return [
+        'submission_id' => $submissionId,
+        'webform_id' => $this->getWebformSubmission($submissionId)
+        ?->getWebform()
+        ?->id(),
+      ];
+    }
+    catch (InvalidPluginDefinitionException | PluginNotFoundException $e) {
```

Comment:
Are these exceptions ever thrown in the `try` block?

Reply:
Removed